### PR TITLE
feat: onboard routes on swagger

### PR DESCRIPTION
**Pull request by Gauravudia**
## Description of the change

> 
Onboard routes on swagger
  - Miscellaneous
  - Destination
  - Bulk Upload
  - Delivery
  - Source
  - User Transform

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: swagger/components/schemas/processorTransformationRequest.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+title: Processor Transformation Request
+type: object
+properties:
+  request:
+    type: object
```
Comment (utsabc) [inline swagger/components/schemas/processorTransformationRequest.yaml:5]: This is not needed its dynamically added in Controller

File: swagger/components/schemas/routerTransform.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+              properties: {}
+            XML:
+              type: object
+              properties: {}
+            FORM:
+              type: object
+              properties: {}
+        files:
+          type: object
+          properties: {}
```
Comment (utsabc) [inline swagger/components/schemas/routerTransform.yaml:42]: Can we put this as schema?

File: swagger/components/schemas/routerTransform.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        metadata:
+          $ref: './metadata.yaml'
```
Comment (utsabc) [inline swagger/components/schemas/routerTransform.yaml:44]: This is not going the be used here (not needed only in delivery response it is used)

File: swagger/components/schemas/processorTransform.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+              type: object
+              properties: {}
+            FORM:
+              type: object
+              properties: {}
+        files:
+          type: object
+          properties: {}
+        metadata:
+          $ref: './metadata.yaml'
```
Comment (utsabc) [inline swagger/components/schemas/processorTransform.yaml:44]: This is not needed here

File: swagger/components/schemas/processorTransform.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+              properties: {}
+            XML:
+              type: object
+              properties: {}
+            FORM:
+              type: object
+              properties: {}
+        files:
+          type: object
+          properties: {}
```
Comment (utsabc) [inline swagger/components/schemas/processorTransform.yaml:42]: Lets put this as schema

File: swagger/routes/testEvent.yaml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                type: array
+                items:
+                  type: object
+                  properties:
+                    message:
+                      type: object
+                    metadata:
+                      type: object
+                    destination:
+                      type: object
```
Comment (utsabc) [inline swagger/routes/testEvent.yaml:33]: add the components for these respective schema

File: swagger/routes/testEvent.yaml [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        required: true
+        schema:
+          type: string
+    responses:
+      '200':
+        description: OK
+        content:
+          '*/*':
+            schema:
+              type: string
```
Comment (utsabc) [inline swagger/routes/testEvent.yaml:91]: do we need this?

File: swagger/components/schemas/routerTransform.yaml (hunk 2 above)
Comment (Gauravudia) [reply swagger/components/schemas/routerTransform.yaml:42]: This file is redundant and not being referenced anywhere. I have removed it.

File: swagger/components/schemas/processorTransform.yaml (hunk 5 above)
Comment (Gauravudia) [reply swagger/components/schemas/processorTransform.yaml:42]: This file is redundant and not being referenced anywhere. I have removed it.

File: swagger/routes/testEvent.yaml (hunk 7 above)
Comment (Gauravudia) [reply swagger/routes/testEvent.yaml:91]: This file is redundant and not being referenced anywhere. I have removed it.

File: swagger/routes/testEvent.yaml (hunk 6 above)
Comment (Gauravudia) [reply swagger/routes/testEvent.yaml:33]: This file is redundant and not being referenced anywhere. I have removed it.

File: swagger/components/schemas/processorTransformationRequest.yaml (hunk 1 above)
Comment (Gauravudia) [reply swagger/components/schemas/processorTransformationRequest.yaml]: Removed.

File: swagger/components/schemas/routerTransform.yaml (hunk 3 above)
Comment (Gauravudia) [reply swagger/components/schemas/routerTransform.yaml]: Remvoed

File: swagger/components/schemas/processorTransform.yaml (hunk 4 above)
Comment (Gauravudia) [reply swagger/components/schemas/processorTransform.yaml]: Removed

File: swagger/components/schemas/processorTransformationRequest.yaml (hunk 1 above)
Comment (Gauravudia) [reply swagger/components/schemas/processorTransformationRequest.yaml]: Should we remove "request" property from `routerTransfromationRequest.yaml` as well

File: swagger/routes/destination.yaml [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+            type: array
+            items:
+              $ref: '../components/schemas/processorTransformationRequest.yaml'
+    responses:
+      200:
+        description: Success
+        content:
+          application/json:
+            schema:
+              $ref: '../components/schemas/processorTransformationResponse.yaml'
```
Comment (utsabc) [inline swagger/routes/destination.yaml:34]: Response is also array right

File: swagger/routes/destination.yaml [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+            $ref: '../components/schemas/routerTransformationRequest.yaml'
+    responses:
+      200:
+        description: Success
+        content:
+          application/json:
+            schema:
+              $ref: '../components/schemas/routerTransformationResponse.yaml'
+      400:
+        description: Failure
```
Comment (utsabc) [inline swagger/routes/destination.yaml:71]: I dont think this is the correct representation of success response can you please check

File: swagger/routes/destination.yaml [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+        application/json:
+          schema:
+            $ref: '../components/schemas/routerTransformationRequest.yaml'
+    responses:
+      200:
+        description: Success
+        content:
+          application/json:
+            schema:
+              $ref: '../components/schemas/routerTransformationResponse.yaml'
```
Comment (utsabc) [inline swagger/routes/destination.yaml:116]: Same here

File: swagger/routes/source.yaml [hunk 11]
```diff
@@ -0,0 +1,39 @@
+/{version}/sources/{source}:
+  post:
+    tags:
+      - Source
+    summary: Source Transformation
+    params:
+      properties:
+        version:
```
Comment (koladilip) [inline swagger/routes/source.yaml:8]: `in: path` is not required?

File: swagger/routes/destination.yaml (hunk 9 above)
Comment (Gauravudia) [reply swagger/routes/destination.yaml]: updated.

File: swagger/routes/destination.yaml (hunk 10 above)
Comment (Gauravudia) [reply swagger/routes/destination.yaml]: updated.

File: swagger/components/schemas/routerTransformationResponse.yaml [hunk 12]
```diff
@@ -0,0 +1,18 @@
+title: Processor Transformation Response
```
[the remark this answers is not in the document]
Comment (Gauravudia) [reply swagger/components/schemas/routerTransformationResponse.yaml]: Router. Updated the title

File: swagger/routes/source.yaml (hunk 11 above)
Comment (Gauravudia) [reply swagger/routes/source.yaml]: It is required. Updated the source.yaml

File: swagger/components/schemas/sourceTransformationResponse.yaml [hunk 13]
```diff
@@ -0,0 +1,13 @@
+title: Source Transformation Response
+type: object
+properties:
+  output:
+    $ref: './sourceTransformationOutput.yaml'
+  error:
+    type: object
```
[the remark this answers is not in the document]
Comment (Gauravudia) [reply swagger/components/schemas/sourceTransformationResponse.yaml:7]: In Index.ts `Object` is mentioned in **error** and **statsTag** property

File: swagger/components/schemas/rudderMessage.yaml [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+  channel:
+    type: string
+  context:
+    type: object
+  originalTimestamp:
+    type: string
+  sentAt:
+    type: string
+  timestamp:
+    type: string
```
[the remark this answers is not in the document]
Comment (Gauravudia) [reply swagger/components/schemas/rudderMessage.yaml:24]: added datetime format.

File: swagger/components/schemas/rudderMessage.yaml [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+  event:
+    type: string
+  integrations:
+    type: object
```
[the remark this answers is not in the document]
Comment (Gauravudia) [reply swagger/components/schemas/rudderMessage.yaml:30]: Don't see any specific strucure we have in `index.ts` file. Added the example in yaml for reference.

**Review by utsabc — APPROVED** [review]

**Comment by utsabc** [comment]
LGTM but do test the swagger api's before merging

**Review by utsabc — APPROVED** [review]

File: swagger/components/examples/batch/failure.yaml [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+statusCode: 400
+error: message type identify is not supported
+statTags:
+  errorCategory: dataValidation
+  errorType: instrumentation
+destination:
+  ID: 1pYpzzvcn7AQ2W9GGIAZSsN6Mfq
+  Name: PINTEREST_TAG
+  Config:
+    tagId: '123456789'
```
Comment (utsabc) [inline swagger/components/examples/batch/failure.yaml:13]: Pinterest tag is router batching

File: swagger/components/examples/batch/success.yaml [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+          event_id: 7208bbb6-2c4e-45bb-bf5b-ad426f3593e9
+          app_id: '429047995'
+          advertiser_id: 'advertiserId123'
+          user_data:
+            em:
+              - 48ddb93f0b30c475423fe177832912c5bcdce3cc72872f8051627967ef278e08
+            ph:
+              - d164bbe036663cb5c96835e9ccc6501e9a521127ea62f6359744928ba932413b
+            ln:
+              - bdfdee6414a89d72bfbf5ee90b1f85924467bae1e3980d83c2cd348dc31d5819
```
Comment (utsabc) [inline swagger/components/examples/batch/success.yaml:24]: Same here

File: swagger/components/examples/batch/failure.yaml (hunk 16 above)
Comment (Gauravudia) [reply swagger/components/examples/batch/failure.yaml]: update with amplitude sample response

File: swagger/components/examples/batch/success.yaml (hunk 17 above)
Comment (Gauravudia) [reply swagger/components/examples/batch/success.yaml]: update with amplitude sample response

**Review by utsabc — APPROVED** [review]

File: swagger/components/schemas/rudderMessage.yaml (hunk 15 above)
[the remark this answers is not in the document]
Comment (koladilip) [reply swagger/components/schemas/rudderMessage.yaml:30]: do we have documentation what all things we support as part of integration object

**Review by koladilip — APPROVED** [review]

File: swagger/components/schemas/rudderMessage.yaml (hunk 15 above)
[the remark this answers is not in the document]
Comment (Gauravudia) [reply swagger/components/schemas/rudderMessage.yaml:30]: This information is found in the [Common Fields documentation](https://www.rudderstack.com/docs/event-spec/standard-events/common-fields/).
`You can specify the destinations for which you want to enable or disable event sending.`

However, What we support in the integration object depend on the `destinations`. In the documentation specific to each destination, we outline the expected content for the integration object for that destination

File: swagger/components/schemas/sourceTransformationResponse.yaml (hunk 13 above)
[the remark this answers is not in the document]
Comment (koladilip) [reply swagger/components/schemas/sourceTransformationResponse.yaml:7]: you can get from code, @utsabc please help

File: swagger/components/schemas/rudderMessage.yaml (hunk 15 above)
[the remark this answers is not in the document]
Comment (koladilip) [reply swagger/components/schemas/rudderMessage.yaml:30]: can we come with type accordingly?

[the remark this answers is not in the document]
Comment (Gauravudia) [reply swagger/components/schemas/rudderMessage.yaml:30]: Structure is not constant here.
E.g.
```
{
    integrations: {
      All: true,
      Mouseflow: {
        customVariables: {
          conversionRate: "100",
          total: 4,
        },
      },
    },
}
```

[the remark this answers is not in the document]
Comment (koladilip) [reply swagger/components/schemas/rudderMessage.yaml:30]: ok got it

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

File: swagger/components/schemas/sourceTransformationResponse.yaml (hunk 13 above)
[the remark this answers is not in the document]
Comment (Gauravudia) [reply swagger/components/schemas/sourceTransformationResponse.yaml]: added statsTag schema

**Review by krishna2020 — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**